src/lib.rs

```diff
@@ -287,6 +292,16 @@ impl<T: Committable> RawCommitmentBuilder<T> {
         self
     }
 
+    pub fn optional<N: Committable>(self, field: &str, o: &Option<N>) -> Self {
+        match o {
+            Some(s) => {
+                let commit = s.commit();
+                self.fixed_size_field(field, &commit.0)
```

Comment:
@ImJeremyHe we should generalize the commit logic with a closure:

```
builder = builder.optional(&self.block_hash, |builder, block_hash| {
  // Closure that gets called if the option is `Some`.
  builder.field("block_hash", block_hash)
});

```

Reply:
I think it is a little strange to use a closure to define a specific way of `Some`. This way I think is more comfortable for the callers. Just like what [serde](https://github.com/serde-rs/serde/blob/8de84b7ca32afd5037326025ed8fcfd06ed9db5f/serde/src/ser/impls.rs#L101) does.